app/controllers/concerns/formatting_concern.rb

```diff
@@ -36,6 +36,13 @@ def pass_records_through_xml_parser(records)
     end
   end
 
+  def valid_barcode(barcode)
+    valid_barcode = false
+    if barcode =~ /^(32101[0-9]{9}|PULTST[0-9]{5})$/
+      valid_barcode = true
+    end
+  end
```

Comment:
This method should not return `nil` when the barcode is invalid.  I think the method can be simplified to:
```
def valid_barcode(barcode='')
  barcode =~ /^(32101[0-9]{9}|PULTST[0-9]{5})$/
end
```